### PR TITLE
Make input commit filtering work for pc list-job.

### PR DESCRIPTION
**Pull request by jdoliner**

### CODE REVIEW

**Comment by jdoliner** [comment]
Fixes #2640

**Comment by brycemcanally** [comment]
Is input commit filtering implemented in the back end? It looks like only pipeline and output commit filtering are currently implemented because the InputCommit field is not being used (unless I am overlooking something). Also, since the input commits is now a flag, wouldn't it be better to change the use to "list-job [-i commits]" or get rid of the commits portion entirely?

**Comment by jdoliner** [comment]
@brycemcanally good point, I think that's been broken since inputs became hierarchical. I'll fix that in this pr as well.

File: src/server/pps/server/api_server.go [hunk 1]
```diff
@@ -477,7 +477,7 @@ func (a *apiServer) InspectJob(ctx context.Context, request *pps.InspectJobReque
 // listJob is the internal implementation of ListJob shared between ListJob and
 // ListJobStream. When ListJob is removed, this should be inlined into
 // ListJobStream.
-func (a *apiServer) listJob(ctx context.Context, pipeline *pps.Pipeline, outputCommit *pfs.Commit) ([]*pps.JobInfo, error) {
+func (a *apiServer) listJob(ctx context.Context, pipeline *pps.Pipeline, outputCommit *pfs.Commit, inputCommits []*pfs.Commit) ([]*pps.JobInfo, error) {
```
Comment (msteffen) [inline src/server/pps/server/api_server.go]: Do we need to call InspectCommit on these? I.e. if someone `pc list-job -p mypipeline -i master` should we try to convert `master` to an ID?

Comment (jdoliner) [reply src/server/pps/server/api_server.go]: That is a good point, we should probably do the same for output commits as well.

**Pull request merged**